src/coreclr/src/vm/eventpipe.cpp

```diff
@@ -963,4 +987,37 @@ bool EventPipe::IsLockOwnedByCurrentThread()
 }
 #endif
 
+// This method will block runtime bring-up IFF DOTNET_DiagnosticsMonitorAddress != nullptr and DOTNET_DiagnosticsMonitorStopOnStart!=0 (it's default state)
+// The s_ResumeRuntimeStartupEvent event will be signaled when the Diagnostics Monitor uses the ResumeRuntime Diagnostics IPC Command
+void EventPipe::PauseForTracingAgent()
+{
+    LIMITED_METHOD_CONTRACT;
+
+    CLRConfigStringHolder pDotnetDiagnosticsMonitorAddress = CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_DOTNET_DiagnosticsMonitorAddress);
+    if (pDotnetDiagnosticsMonitorAddress != nullptr)
+    {
+        DWORD dwDotnetDiagnosticsMonitorStopOnStart = CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_DOTNET_DiagnosticsMonitorStopOnStart);
+        if (dwDotnetDiagnosticsMonitorStopOnStart != 0)
+        {
+            s_ResumeRuntimeStartupEvent = new CLREventStatic();
+            s_ResumeRuntimeStartupEvent->CreateManualEvent(false);
+            const DWORD dwFiveSecondWait = s_ResumeRuntimeStartupEvent->Wait(5000, false);
+            if (dwFiveSecondWait == WAIT_TIMEOUT)
+            {
+                STRESS_LOG0(LF_DIAGNOSTICS_PORT, LL_ALWAYS, "The runtime has been configured to pause during startup and is awaiting a Diagnostics IPC ResumeStartup command.");
```

Comment:
Maybe I am being too bold, but I was thinking of writing this warning with printf (also logging it to stress log is good too).

Reply:
I tried using `fprintf(stdout, <msg>)`, but there seemed to be some kind of caching going on and the message was never flushed to the console until process shutdown.